roles/container_runtime/templates/crio.conf.j2

```diff
@@ -141,16 +141,18 @@ signature_policy = ""
 # The valid values are mkdir and ignore.
 image_volumes = "mkdir"
 
+# CRI-O reads its configured registries defaults from the containers/image configuration
+# file, /etc/containers/registries.conf. Modify registries.conf if you want to
+# change default registries for all tools that use containers/image.  If you
+# want to modify just crio, you can change the registies configuration in this
+# file.
+
 # insecure_registries is used to skip TLS verification when pulling images.
-insecure_registries = [
-{{ l_insecure_crio_registries|default("") }}
-]
+# insecure_registries = []
```

Comment:
Is this going to be a problem for users that upgrade?  We don't overwrite config files by default, so anyone that already has crio deployed will never get this change unless we create a task to do so.

Reply:
So if `insecure_registries` is not commented out in crio.conf, cri-o will pick up the registries from there. If it is commented out then it checks /etc/containers/registries.conf. So even if the users upgrade and their config file doesn't change they shouldn't have any issues, but in this case they will as we removed `l_insecure_crio_registries` from the yaml. I would recommend asking them to update the config files to match where cri-o is at fully.
I can put the crio_registries variables back in if we want to ensure we don't break users who don't upgrade the config files.

Reply:
@mrunalp @michaelgugino should I leave in the crio_registries stuff to not break user who don't upgrade their config files?

Reply:
If we don't support insecure registries in the other file, we should leave it in-place in this file.  Relying on users to make manual config changes to their crio configs is not a good upgrade process.  There could be hundreds of hosts to migrate.  Whatever changes are required, we need to handle that for them.

Reply:
Sorry I wasn't clear before. The insecure and additional registries are supported in the /etc/containers/registries.conf file by cri-o. But cri-o also lets you have that defined in the crio.conf file in the event you want to configure the registries for cri-o only and not other tools using the /etc/containers/registries.conf file as well.
When users upgrade, but don't upgrade their config files, cri-o will still work fine as it will still be picking up the registries from the crio.conf file instead of the /etc/containers/registries.conf file. If we remove the `l_insecure_crio_registries` and `l_additional_crio_registries` variables from the code it will cause issues as the crio.conf will still be pointing at the values of those but won't be able to find anything.
So leaving `l_insecure_crio_registries` and `l_additional_crio_registries` in main.yml looks to be the best option for users who upgrade but don't upgrade their config files.

Reply:
Leaving both in place.